openssl/posix.mk

```diff
@@ -51,7 +55,7 @@ USE_FETCH_HISTORY = 1
 endif
 
 ifndef USE_CRYPTO_API
-USE_CRYPTO_API = 0
+USE_CRYPTO_API = 1
 endif
```

Comment:
@Xavrax in [current file](https://github.com/pubnub/c-core/blob/master/openssl/posix.mk#L1) (at the end of the line) you can see that crypto files were there unconditionally and when gurads has been fixed it started to cause problems. 
Here it has been enabled by default because CI runs `all` task which require crypto module to be used - this is the reason why I enabled it in those files.